app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/bugreport/NetworkTypeCollector.kt

```diff
@@ -167,12 +179,19 @@ class NetworkTypeCollector @Inject constructor(
         vpnStopReason: VpnStopReason,
     ) {
         (context.applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?)?.let {
+            // always unregistered, even if not previously registered
             it.safeUnregisterNetworkCallback(wifiNetworkCallback)
             it.safeUnregisterNetworkCallback(cellularNetworkCallback)
             it.safeUnregisterNetworkCallback(privateDnsCallback)
         }
     }
 
+    private suspend fun shouldSkipVpnRestart(): Boolean {
```

Comment:
NIT: Confusing name here because when this returns true, we actually restart the VPN and not skip it.